extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ColumnChunkReaderImpl.java

```diff
@@ -294,58 +287,51 @@ private static org.apache.parquet.format.Encoding getEncoding(PageHeader pageHea
         }
     }
 
+    private PageHeader readPageHeader(final SeekableByteChannel ch) throws IOException {
+        try (final InputStream in = SeekableChannelsProvider.channelPositionInputStream(channelsProvider, ch)) {
+            return Util.readPageHeader(in);
+        }
+    }
+
     private static int getNumValues(PageHeader pageHeader) {
         return pageHeader.isSetData_page_header()
                 ? pageHeader.getData_page_header().getNum_values()
                 : pageHeader.getData_page_header_v2().getNum_values();
     }
 
-    private final class ColumnPageReaderIteratorIndexImpl implements ColumnPageReaderIterator {
```

Comment:
To give you some context behind this class, we need to understand the two page reader implementations which are used for accessing any page in the file:
- `VariablePageSizeColumnChunkPageStore`: To access page i, this class iterates through all the pages leading upto i and builds a reader objects.
Internally, this class can use two implementations of ColumnPageReaderIterator:
    - `ColumnPageReaderIteratorImpl`: This class will read the page header for each page, decides whether each page needs a dictionary and then finally builds and returns reader objects.
    - `ColumnPageReaderIteratorIndexImpl`: This class uses offset index to locate the pages and thus does not need read the page header. It can directly find offset of each page in the file and build a reader object.
- `OffsetIndexBasedColumnChunkPageStore`: To access page i, this class directly uses offset index to locate the page in the file and builds a reader objects. Thus no iterating needed.

So now your question might be, when do we use each of them:
- If the parquet file has offset index and doesn't have the offset-index bug from deephaven version below 0.31.0, then we use `OffsetIndexBasedColumnChunkPageStore` (check `ColumnChunkPageStore::canUseOffsetIndexBasedPageStore` for more details).
- Else we use `VariablePageSizeColumnChunkPageStore`
      - If the parquet file has offset index, we use `ColumnPageReaderIteratorIndexImpl`
      - If the parquet file doesn't have offset index, we use `ColumnPageReaderIteratorImpl`

So overall `ColumnPageReaderIteratorIndexImpl` is used to handle parquet files written by deephaven version below 0.31.0 with buggy offset index information.
Note that these files can also be read by `ColumnPageReaderIteratorImpl` because it doesn't even use the offset index information. So is not affected by the bug. But we used `ColumnPageReaderIteratorIndexImpl` because it was more efficient in that it didn't need to read the page header.

But now, after this new bug, we know that we always need to read the page header to decide dictionary encoding of each page. Therefore, I decided to remove `ColumnPageReaderIteratorIndexImpl` because now there is no performance advantage over `ColumnPageReaderIteratorImpl`.

And this helped me make more improvements in how we compute data offset and numValues. Basically, now since we are always reading page header up front, we can directly compute the offset of data in the page and the number of values in the page.